### ISSUE TITLE
check_mysql_query: Use after free resulting in SEGFAULT with musl libc

### ISSUE THREAD

**Issue by bratkartoffel** [nagios-plugins/nagios-plugins#602]
Hi there,

the plugin crashes on alpine with musl libc as the resul row is accessed after the resultset has been freed and the connection closed.
```c
138         /* fetch the first row */
139         if ( (row = mysql_fetch_row (res)) == NULL) {
140                 error = strdup(mysql_error(&mysql));
141                 mysql_free_result (res);
142                 mysql_close (&mysql);
143                 die (STATE_CRITICAL, "QUERY %s: Fetch row error - %s\n", _("CRITICAL"), error);
144         }
145
146         /* free the result */
147         mysql_free_result (res);
148
149         /* close the connection */
150         mysql_close (&mysql);
151
152         if (! is_numeric(row[0])) {
153                 die (STATE_CRITICAL, "QUERY %s: %s - '%s'\n", _("CRITICAL"), _("Is not a numeric"), row[0]);
154         }
155
156         value = strtod(row[0], NULL);
```

Please move the free and close from lines 147 / 150 after any access to the returned result (currently line 156).
Issue at alpine: https://gitlab.alpinelinux.org/alpine/aports/-/issues/12601

Thanks,
Simon

**Comment by instantlinux** [comment]
Any update? This bug is critical: it makes the plugin unusable.

**Comment by bratkartoffel** [comment]
@instantlinux for alpine we've [packaged a patch](https://git.alpinelinux.org/aports/commit/?id=553b52b33f4111f77c68ac4f2d42c6763e561527) to fix this issue (in 2.3.2-r2) , so using alpine:edge should be fine for now. Nevertheless, the issue should be fixed upstream

**Comment by sawolf** [comment]
Thanks for reporting this - I've added your patch to `master` and it'll go in the next release.
